client/app/components/Link.tsx

```diff
@@ -1,41 +1,28 @@
 import React from "react";
 import Button, { ButtonProps as AntdButtonProps } from "antd/lib/button";
+import { withIcon, WithIconProps } from "./WithIcon";
 
 function DefaultLinkComponent({ children, ...props }: React.AnchorHTMLAttributes<HTMLAnchorElement>) {
   return <a {...props}>{children}</a>;
 }
 
 Link.Component = DefaultLinkComponent;
 
-interface LinkProps extends Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, "role" | "type" | "target"> {
+interface LinkProps extends Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, "role" | "type"> {
   href: string;
+  target?: never; // use external link
 }
 function Link({ children, ...props }: LinkProps) {
   return <Link.Component {...props}>{children}</Link.Component>;
 }
 
-interface LinkWithIconProps extends LinkProps {
-  children: string;
-  icon: JSX.Element;
-  alt: string;
-  target?: "_self" | "_blank" | "_parent" | "_top";
-}
-
-function LinkWithIcon({ icon, alt, children, ...props }: LinkWithIconProps) {
-  return (
-    <Link.Component {...props}>
-      {children} {icon} <span className="sr-only">{alt}</span>
-    </Link.Component>
-  );
-}
-
-Link.WithIcon = LinkWithIcon;
+Link.WithIcon = withIcon(Link.Component);
```

Comment:
There is a little problem with this line. `Link.Component` is used to override link behavior. Extensions can do sth like:

```jsx
import Link from "...";

function CustomLinkComponent(...) {
  return ...;
}

Link.Component = CustomLinkComponent;
```

But since `Link.WithIcon` is initialized when `Link` component is imported - `Link.WithIcon` will capture default `Link.Component` and overriding it will not have any effect when using `Link.WithIcon`